spec/lib/logstash/java_integration_spec.rb

```diff
@@ -0,0 +1,82 @@
+# encoding: utf-8
+require "spec_helper"
+require "logstash/java_integration"
+
+describe "Java integration" do
+
+  context "type equivalence" do
+
+    it "ArrayList should report to be a Ruby Array" do
+      expect(Java::JavaUtil::ArrayList.new.is_a?(Array)).to be_true
+    end
+
+    it "ArrayList should be class equivalent to Ruby Array" do
+      expect do
+        case Java::JavaUtil::ArrayList.new
+        when Array
+          true
+        else
+          raise
+        end
+      end.not_to raise_error
+    end
+
+    it "LinkedHashMap should report to be a Ruby Hash" do
+      expect(Java::JavaUtil::LinkedHashMap.new.is_a?(Hash)).to be_true
+    end
+
+    it "LinkedHashMap should be class equivalent to Ruby Hash" do
+      expect do
+        case Java::JavaUtil::LinkedHashMap.new
+        when Hash
+          true
+        else
+          raise
+        end
+      end.not_to raise_error
+    end
+  end
+
+  context "Java::JavaUtil::Collection" do
+
+    it "should delete a single instance" do
+      a = Java::JavaUtil::ArrayList.new
+      a += ["foo", "bar"]
+      result = a.delete("foo")
+      expect(result).to eq("foo")
+      expect(a.size).to eq(1)
+      expect(a.first).to eq("bar")
+    end
+
+    it "should delete a multiple instances" do
+      a = Java::JavaUtil::ArrayList.new
+      a += ["foo", "bar", "foo"]
+      result = a.delete("foo")
+      expect(result).to eq("foo")
+      expect(a.size).to eq(1)
+      expect(a.first).to eq("bar")
+    end
+
+    it "should return nil when nothing to delete" do
+      a = Java::JavaUtil::ArrayList.new
+      a += ["foo", "bar", "foo"]
+      result = a.delete("baz")
+      expect(result).to be_nil
+      expect(a.size).to eq(3)
+
+      a = Java::JavaUtil::ArrayList.new
+      result = a.delete("baz")
+      expect(result).to be_nil
+      expect(a.size).to eq(0)
+   end
+
+    it "should yeild block when nothing to delete" do
+      a = Java::JavaUtil::ArrayList.new
+      a += ["foo", "bar", "foo"]
+      result = a.delete("baz") {"boo"}
+      expect(result).to eq("boo")
+      expect(a.size).to eq(3)
+    end
+  end
+
+end
```

Comment:
I don't think it is over engineering:

- `expect_change` is actually a better test here than asserting an exact value.
- Running a simple expectation is fast and isolated http://betterspecs.org/#single
- Line of code from your examples and mine are really close similar.
- Using subject and let make test code easier to refactor, move or reuse via shared helpers.